### PR TITLE
Add chat notifies

### PR DESCRIPTION
**Pull request by fedddddd**

### CODE REVIEW

**Review by momo5502 — CHANGES_REQUESTED** [review]
The say command has already been registered in party.cpp
I suggest hooking the server command handler instead.
Also not sure if you are calling script api functions from the right thread.

**Review by momo5502 — CHANGES_REQUESTED** [review]
That's not what I meant by hooking the handler.
Just check how it was done in IW4x: https://github.com/XLabsProject/iw4x-client/blob/a2a303c3d0500c1f0bde8250ae375fd747e8b6e4/src/Components/Modules/Dedicated.cpp#L308

File: src/client/component/logfile.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+			return hidden;
+		}
+	}
+
+	const auto say_stub = utils::hook::assemble([](utils::hook::assembler& a)
+	{
+		const auto hidden = a.newLabel();
+
+		a.call(0x1404F63C0);
```
Comment (momo5502) [inline src/client/component/logfile.cpp:41]: Use call_aligned instead here.
The reason is the stack needs to be 0x10 aligned before a call and you have to reserve a stack frame for the callee. The call_aligned macro does that automatically for you.

File: src/client/component/logfile.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+		a.mov(rdx, rdi);
+		a.mov(rcx, rbx);
+
+		a.call(evaluate_say);
```
Comment (momo5502) [inline src/client/component/logfile.cpp:46]: Same here

File: src/client/component/logfile.cpp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+namespace logfile
+{
+	namespace
+	{
+		bool evaluate_say(char* text, game::mp::gentity_s* ent)
+		{
+			auto hidden = false;
+
+			const auto level = scripting::entity(*game::levelEntityId);
```
Comment (momo5502) [inline src/client/component/logfile.cpp:19]: Scripting interaction strictly requires this code to be executed on the server thread. Is this the case? Did you verify it? You can do so by debugging the code and setting a breakpoint here. Then, in Visual Studio, check the name of the thread this function was called, it should be called "Server".

**Review by momo5502 — CHANGES_REQUESTED** [review]
A few small things to change, otherwise this pull request starts looking very good 👍

File: src/client/component/logfile.cpp (hunk 3 above)
Comment (fedddddd) [reply src/client/component/logfile.cpp:19]: Yes it is the server thread

File: src/client/component/logfile.cpp (hunk 1 above)
Comment (fedddddd) [reply src/client/component/logfile.cpp:41]: Good to know thanks

**Review by momo5502 — APPROVED** [review]

**Pull request merged**